clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/mapreduce/datawriter/jobs/DataWriterMRJob.java

```diff
@@ -203,6 +206,11 @@ private void setupDefaultJobConf(JobConf conf, PushJobSetting pushJobSetting, Ve
         props.getString(ZSTD_COMPRESSION_LEVEL, String.valueOf(Zstd.maxCompressionLevel())));
     conf.setBoolean(ZSTD_DICTIONARY_CREATION_SUCCESS, pushJobSetting.isZstdDictCreationSuccess);
 
+    // Deterministic producerGUID generator
+    UUID producerGuid = UUID.randomUUID();
+    conf.setLong(PUSH_JOB_GUID_MOST_SIGNIFICANT_BITS, producerGuid.getMostSignificantBits());
+    conf.setLong(PUSH_JOB_GUID_LEAST_SIGNIFICANT_BITS, producerGuid.getLeastSignificantBits());
```

Comment:
I don't understand this... random is not deterministic, or is it?

Reply:
It is random - so, it helps to guarantee it to be unique across different VPJ jobs. But because we generate this random UUID once, and use that during the execution, it will be "deterministic" for all tasks of that compute job.

Reply:
I updated the comments around this to make it more explanatory

Reply:
Thanks, the improved comment is good.